### PR TITLE
boot: support device tree for syslinux boot

### PR DESCRIPTION
**Pull request by yurinnick**
- Lookup for `fdt` term in syslinux configs
- Load device tree file to the end of initrd

### CODE REVIEW

**Comment by yurinnick** [comment]
I'm quite confused about syslinux tests, removing `DeviceTree` from the expected struct doesn't fail the test. Am I missing something?

**Comment by yurinnick** [comment]
Also, what's the purpose of `LinuxImage.Label()`? Compared to `LinuxImage.String()` it's almost identical.

**Comment by rminnich** [comment]
where are we with this? I think I owe you something but ...

**Comment by yurinnick** [comment]
Tested only on pinephone:
- `fdt` parameter works just fine
- `fdtdir` not implemented and I don't have a good idea on how tackle it rn

I stopped on trying to push code coverage to acceptable level, is there a way to run this stat locally? I though I covered the new lines, but looks like it's not the case.

Overall, I am a little concerned that `kexec` with `fdt` works a little flaky at times: every once in a while I get black screen like it device tree wasn't loaded.

**Comment by yurinnick** [comment]
>  I think I owe you something but ...

What do you mean?

**Comment by yurinnick** [comment]
Update:
- `fdt` works, I tested it with Mobian and Archlinux and both worked after I replaced `fdtdir` with `fdt` in extlinux config
- I am going to add `fdt` option to kexec in a separate PR
- There's a small patch code coverage miss (73.33% of diff hit with target 73.41%) due to completely missing tests for `LinuxImage.Load()` method

If you don't mind, could you answer the questions above, it would help me in my future contributions. Thanks!

File: pkg/boot/linux.go [hunk 1]
```diff
@@ -167,6 +197,9 @@ func (li *LinuxImage) Load(verbose bool) error {
 			log.Printf("Initrd: %s", i.Name())
 		}
 		log.Printf("Command line: %s", li.Cmdline)
+		if li.DeviceTree != nil {
+			log.Print("Device tree loaded: true")
```
[the remark this answers is not in the document]
Comment (rminnich) [reply pkg/boot/linux.go:201]: OMG, the whole package does this. OK, I'm going to approve this PR.

**Review by rminnich — APPROVED** [review]

**Pull request merged**